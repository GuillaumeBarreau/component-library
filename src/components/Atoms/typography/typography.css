/* Typography */
@font-face {
  font-family: "lato";
  src: local("lato"),
    url(../../_settings/fonts/Lato/Lato-Regular.ttf) format("truetype");
}

@font-face {
  font-family: "lato-bold";
  src: local("bato-bold"),
    url(../../_settings/fonts/Lato/Lato-Bold.ttf) format("truetype");
}

@font-face {
  font-family: "lato-italic";
  src: local("bato-italic"),
    url(../../_settings/fonts/Lato/Lato-Italic.ttf) format("truetype");
}

@font-face {
  font-family: "lato-black";
  src: local("bato-black"),
    url(../../_settings/fonts/Lato/Lato-Black.ttf) format("truetype");
}

@font-face {
  font-family: "lato-light";
  src: local("bato-light"),
    url(../../_settings/fonts/Lato/Lato-Light.ttf) format("truetype");
}

@font-face {
  font-family: "lato-thin";
  src: local("bato-thin"),
    url(../../_settings/fonts/Lato/Lato-Thin.ttf) format("truetype");
}

.typography {
  font-family: lato, Arial, serif;
}

.typography-bold {
  font-family: lato-black, Arial, serif;
}

.typography-italic {
  font-family: lato-italic, Arial, serif;
}

.typography-black {
  font-family: lato-black, Arial, serif;
}

.typography-light {
  font-family: lato-light, Arial, serif;
}

.typography-xsmall {
  font-size: 12px;
}

.typography-small {
  font-size: 16px;
}

.typography-default {
  font-size: 18px;
}

.typography-medium {
  font-size: 20px;
}

.typography-large {
  font-size: 24px;
}

.typography-xlarge {
  font-size: 32px;
}
