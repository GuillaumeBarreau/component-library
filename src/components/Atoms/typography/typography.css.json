{"typography":"_typography_iie5v_32","typography-bold":"_typography-bold_iie5v_36","typography-italic":"_typography-italic_iie5v_40","typography-black":"_typography-black_iie5v_44","typography-light":"_typography-light_iie5v_48","typography-xsmall":"_typography-xsmall_iie5v_52","typography-small":"_typography-small_iie5v_56","typography-medium":"_typography-medium_iie5v_60","typography-large":"_typography-large_iie5v_64","typography-xlarge":"_typography-xlarge_iie5v_68"}