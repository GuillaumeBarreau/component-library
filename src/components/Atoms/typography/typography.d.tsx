export type TSizes =
  | "xsmall"
  | "small"
  | "medium"
  | "large"
  | "xlarge"
  | "default";

export type TVariants =
  | "h1"
  | "h2"
  | "h3"
  | "h4"
  | "h5"
  | "h6"
  | "p"
  | "span"
  | undefined;

export interface ComponentTypographyProps {
  className?: string;
  font?: string;
  size?: TSizes;
  children: string;
  variant?: TVariants;
}
