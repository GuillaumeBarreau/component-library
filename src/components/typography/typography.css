/* Typography */
@font-face {
   font-family: 'lato';
   src: local('lato'), url(../../fonts/Lato/Lato-Regular.ttf) format('truetype');
}

@font-face {
   font-family: 'lato-bold';
   src: local('bato-bold'), url(../../fonts/Lato/Lato-Bold.ttf) format('truetype');
}

@font-face {
   font-family: 'lato-italic';
   src: local('bato-italic'), url(../../fonts/Lato/Lato-Italic.ttf) format('truetype');
}

@font-face {
   font-family: 'lato-black';
   src: local('bato-black'), url(../../fonts/Lato/Lato-Black.ttf) format('truetype');
}

@font-face {
   font-family: 'lato-light';
   src: local('bato-light'), url(../../fonts/Lato/Lato-Light.ttf) format('truetype');
}

@font-face {
   font-family: 'lato-thin';
   src: local('bato-thin'), url(../../fonts/Lato/Lato-Thin.ttf) format('truetype');
}

.typography {
   font-family: lato, Arial, serif;
}

.typography-bold {
   font-family: lato-black, Arial, serif;
}

.typography-italic {
   font-family: lato-italic, Arial, serif;
}

.typography-black {
   font-family: lato-black, Arial, serif;
}

.typography-light {
   font-family: lato-light, Arial, serif;
}

.typography-xsmall {
   font-size: 12px;
}

.typography-small {
   font-size: 16px;
}

.typography-medium {
   font-size: 20px;
}

.typography-large {
    font-size: 24px;
}

.typography-xlarge {
    font-size: 32px;
}

 
