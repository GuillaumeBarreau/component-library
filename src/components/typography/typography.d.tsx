export type TSizes = "xsmall" | "small" | "medium" | "large" | "xlarge";
export type TVariants = "h1" | "h2" | "h3" | "h4" | "h5" | "h6" | "p" | undefined;

export interface ComponentTypographyProps {
  className?: string;
  font?: string;
  size?: TSizes;
  children: string;
  variant?: TVariants
}