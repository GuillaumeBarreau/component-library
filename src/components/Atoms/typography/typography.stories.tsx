import React from "react";
import {Typography} from "./typography";
import { Meta } from "@storybook/react/types-6-0";
import { ComponentTypographyProps, TSizes } from "./typography.d";
import "../../_settings/colors/colors.css";

export default {
  component: Typography,
  title: "Typography",
  argTypes: {
    className: {
      description: "className content",
      control: "text",
    },
    children: {
      description: "Text content",
    },
  },
  parameters: {
    a11y: {
      // optional selector which element to inspect
      element: "#root",
      // optional flag to prevent the automatic check
      manual: false,
    },
  },
} as Meta;

const sizes:TSizes[] = ["xsmall", "small", "medium", "large", "xlarge"];

const ListTemplate: any = (args: Partial<ComponentTypographyProps>) => (
  <>
    {sizes.map((size:TSizes, index) => (
      <div key={size}>
        {index !== 0 ? <hr></hr> : null}
        <Typography size={"small"}>{size}</Typography>
        <Typography size={size} font={args.font}>{args.children}</Typography>
      </div>
    ))}
  </>
);

export const TypographyDefault = ListTemplate.bind({});
TypographyDefault.args = {
  children: "Typography",
};

export const TypographyBold = ListTemplate.bind({});
TypographyBold.args = {
  children: "Typography",
  font: "bold",
};

export const TypographyLight = ListTemplate.bind({});
TypographyLight.args = {
  children: "Typography",
  font: "light",
};

export const TypographyItalic = ListTemplate.bind({});
TypographyItalic.args = {
  children: "Typography",
  font: "italic",
};

export const TypographyBlack = ListTemplate.bind({});
TypographyBlack.args = {
  children: "Typography",
  font: "black",
};
