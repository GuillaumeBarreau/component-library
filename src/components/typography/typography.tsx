import React from 'react';
import classNames from 'classnames';
import { ComponentTypographyProps } from './typography.d';
import "./typography.css";

const Typography: React.FC<ComponentTypographyProps> = (props) => {

    const { className, font, children, size = "medium", variant = "p" } = props;
    const prefixCls: string = 'typography';

    const classes = classNames(
        prefixCls,
        {
            [`${prefixCls}-${font}`]: font,
            [`${prefixCls}-${size}`]: size,
        },
        className,
    );

    const typographyNode = React.createElement(
        variant,
        {
            className: classes,
        },
        children
    );


    return typographyNode;
};

export default Typography;