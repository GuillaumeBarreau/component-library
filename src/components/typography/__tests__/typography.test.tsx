import React, { Component } from 'react';
import { mount, shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import Typography from '../typography';
import { ComponentTypographyProps } from '../typography.d';

describe('Button', () => {
  let defaultProps: JSX.IntrinsicAttributes & ComponentTypographyProps;

  beforeEach(() => {
    defaultProps = {
      className: "",
      font: "light",
      size: "medium",
      children: "Typography"
    }
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <Typography {...defaultProps}/>
    );
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('mount correctly', () => {
    const wrapper = shallow(
      <Typography {...defaultProps} />
    );
    expect(() => wrapper).not.toThrow();
  });

  it('to contain defaultProps:children correctly', () => {
    const wrapper = shallow(
      <Typography {...defaultProps} />
    );
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain('Typography');
  });

  it('to contain defaultProps:size correctly', () => {
    const wrapper = shallow(
      <Typography {...defaultProps} />
    );
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain('typography-medium');
  });

  it('to contain defaultProps:font correctly', () => {
    const wrapper = shallow(
      <Typography {...defaultProps} />
    );
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain('typography-light');
  });

});