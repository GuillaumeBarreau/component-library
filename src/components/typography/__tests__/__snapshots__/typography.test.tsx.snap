// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
"<p className=\\"typography typography-light typography-medium\\">
  Typography
</p>"
`;
